psi4/src/psi4/libpe/psipe.cc

```diff
@@ -0,0 +1,259 @@
+        cppe_state_.es_operator_copy().save("V_es_psi4.txt",  arma::raw_ascii);
+    }
+    
+    SharedMatrix D = Dm;
+    if (subtract_scf_density && type == PeState::CalcType::electronic_only) {
+        D->subtract(D_scf_);
+    }
+    cppe_state_.update_energies(arma::mat(D->get_pointer(), D->nrow(), D->ncol(), true));
+    
+    size_t n_sitecoords = 3*cppe_state_.get_polarizable_site_number();
+    
+    // induction operator
+    auto V_ind = std::make_shared<Matrix>("V_ind", basisset_->nbf(), basisset_->nbf());
+    if (n_sitecoords) {
+        int current_polsite = 0;
+        arma::vec elec_fields(n_sitecoords, arma::fill::zeros);
+        for (auto& p : potentials_) {
+            if (!p.is_polarizable()) continue;
+            Vector3 site(p.m_x, p.m_y, p.m_z);
+            Vector elec_fields_s = int_helper_.compute_field(site, D);
+            elec_fields[current_polsite*3] = elec_fields_s.get(0);
+            elec_fields[current_polsite*3 + 1] = elec_fields_s.get(1);
+            elec_fields[current_polsite*3 + 2] = elec_fields_s.get(2);
+            current_polsite += 1;
+        }
+        // std::cout << "elec fields : " << n_sitecoords << std::endl << elec_fields << std::endl;
+        cppe_state_.update_induced_moments(elec_fields, iteration, type == PeState::CalcType::electronic_only);
+        arma::vec induced_moments = cppe_state_.get_induced_moments();
+        
+        current_polsite = 0;
+        for (auto& p : potentials_) {
+            if (!p.is_polarizable()) continue;
+            Vector3 site(p.m_x, p.m_y, p.m_z);
+            SharedMatrix V_ind_s = int_helper_.compute_field_integrals(site,
+                                                                       induced_moments.subvec(3*current_polsite,
+                                                                                              3*current_polsite+2));
+            V_ind->add(V_ind_s);
+            current_polsite += 1;
+        }
+    }
+    
+    if (type == PeState::CalcType::total) {
+        V_ind->add(V_es_);
+        D_scf_ = D;
+    }
+    // TODO: only for debugging
+    // cppe_state_.print_summary();
+    
+    if (type == PeState::CalcType::total) iteration++;
+    
+    double energy_result = (type == PeState::CalcType::total ? cppe_state_.get_current_energies().get_total_energy() :
+                                                              cppe_state_.get_current_energies().get("Polarization/Electronic"));
+    return std::pair<double, SharedMatrix>(energy_result, V_ind);
+}
+
+void PeState::print_energy_summary() {
+    cppe_state_.print_summary();
+}
+
+SharedMatrix PeIntegralHelper::compute_multipole_potential_integrals(Vector3 site, int order, std::vector<double>& moments) {
+    auto integrals = std::make_shared<IntegralFactory>(basisset_, basisset_, basisset_, basisset_);
+    std::shared_ptr<OneBodyAOInt> multipole_integrals = static_cast<std::shared_ptr<OneBodyAOInt>>(integrals->ao_efp_multipole_potential());
```

Comment:
I think `auto multipole_integrals = std::make_shared<OneBodyAOInt>(integrals->ao_efp_multipole_potential());` should suffice here.

Reply:
Done in b9742ad4d.